typed-racket-test/fail/pr13289.rkt

```diff
@@ -1,5 +1,5 @@
 #;
-(exn-pred #rx"fail/pr13289.rkt:9:10:.*in: Natural")
+(exn-pred #rx"fail(/|\\\\)pr13289.rkt:9:10:.*in: Natural")
```

Comment:
```
(exn-pred (regexp (format "~a:9:10:.*in: Natural" (build-path "fail" "pr13289.rkt"))))
```
This might be better

Reply:
Unfortunatly it doesn't work because then you get `#rx"(fail\\pr132..."` instead of `#rx"(fail\\\\pr132..."`

Reply:
How about:
```
(string-append
 (regexp-quote (format "~a:9:10:" (build-path "fail" "pr13289.rkt")))
 ".*in: Natural")
```
output: `"fail/pr13289\\.rkt:9:10:.*in: Natural"`

Reply:
How about we avoid checking for the path separator entirely? Something like `"fail.*:9:10:.*in: Natural"`.

Reply:
I don't know about dropping the filename. If the test is for checking the right source location, is the filename not more important than that there was the word "fail" somewhere in the message.
Is the `(/|\\\\)` so bad, if so I can change it to @shhyou's solution.

Reply:
Sorry, I didn't mean to drop the filename there. I'm suggesting `"fail.*pr13289.rkt:9:10:.*in: Natural

Reply:
oh right. And we can use just `.` since `/` and `\\` are both only one caracter.